### PR TITLE
Mecha bump code fix

### PR DESCRIPTION
**Pull request by MadmanMartian**
Now they can properly get atomized by bumping into supermatter
:cl:
 * bugfix: Mechs properly call bump and such now

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -413,88 +413,76 @@
...
+		src.throwing = 0//so mechas don't get stuck when landing after being sent by a Mass Driver
+		src.crashing = null
+
+	if(breakthrough)
+		if(crashing && !istype(crashing,/turf/space))
+			spawn(1)
+				src.throw_at(crashing, 50, src.throw_speed)
+		else
+			spawn(1)
+				crashing = get_distant_turf(get_turf(src), dir, 3)//don't use get_dir(src, obstacle) or the mech will stop if he bumps into a one-direction window on his tile.
```
Comment (jknpj) [inline code/game/mecha/mecha.dm:484]: why not just put that if there?

**Comment by DrSnips** [comment]
While most mechs getting sharded seems fine what about clarke? With it being rather quick I think it should get a pass with shard stuff. I can't say I've even seen someone use this interaction for antag reasons so its more for engineering QoL.

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/game/mecha/mecha.dm:484]: I have no idea, this is deitycode

**Comment by SonixApache** [comment]
just clamp the shard
or even better, don't go near it

**Pull request merged**